internal/generator/vector/output/common/throttle.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import (
+	"fmt"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/generator/framework"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/normalize"
+)
+
+var (
+	SinkTransformThrottle = "sink_throttle"
+
+	UserDefinedSinkThrottle = fmt.Sprintf(`%s_%%s`, SinkTransformThrottle)
+)
+
+func AddThrottleForSink(spec *logging.OutputSpec, inputs []string) []framework.Element {
+	id := fmt.Sprintf(UserDefinedSinkThrottle, spec.Name)
+	return NewThrottleForSink(id, spec, inputs)
+}
+
+func NewThrottleForSink(id string, spec *logging.OutputSpec, inputs []string) []framework.Element {
```

Comment:
Not used outside of package, too

Reply:
Moved to 'normalize' and made more generic